deltachat-ios/Handler/DeviceContactsHandler.swift

```diff
@@ -18,35 +18,50 @@ class DeviceContactsHandler {
     }
 
     private func addContactsToCore() {
-        let storedContacts = fetchContactsWithEmailFromDevice()
-        let contactString = makeContactString(contacts: storedContacts)
-        dc_add_address_book(mailboxPointer, contactString)
-        contactListDelegate?.deviceContactsImported()
+        fetchContactsWithEmailFromDevice() { contacts in
+            DispatchQueue.main.async {
+                let contactString = self.makeContactString(contacts: contacts)
+                dc_add_address_book(mailboxPointer, contactString)
```

Comment:
you could move dc_add_address_book to the Wrapper.swift. I tried to ged rid of these unwrapped calls in the past when I was working on code that used them.

Reply:
fixed